### PR TITLE
Add flag to choose template

### PR DESCRIPTION
**Pull request by j4b6ski**
This change allows providing a custom template arg flag -t/--template:
```
sinol-make init foo -t [repo link or disk path] [optional subdir]
```
It defaults to the upstream repository and example_package subdir.

Changed the example_package id pattern from `abc` to `__ID__`. All filenames and file contents are now substituted.

Changed the positional output directory argument to `-o/--output` flag to help avoid accidents.

Added result directory cleanup on failure.

### CODE REVIEW

File: src/sinol_make/commands/init/__init__.py [hunk 1]
```diff
@@ -23,21 +27,35 @@ def configure_subparser(self, subparser: argparse.ArgumentParser):
...
+        subdir = template_paths[1] if len(template_paths) > 1 else ''
+
         self.used_tmpdir = tempfile.TemporaryDirectory()
         tmp_dir = self.used_tmpdir.name
-        ret = subprocess.run(['git', 'clone', '-q', '--depth', '1', repo], cwd=tmp_dir)
-        if ret.returncode != 0:
-            util.exit_with_error("Could not access repository. Please try again.")
-        path = os.path.join(tmp_dir, package_dir)
+
+        is_url = template.startswith("https://") or template.startswith("git@")
```
Comment (MasloMaslane) [inline src/sinol_make/commands/init/__init__.py:47]: What about `ssh://`?

File: src/sinol_make/commands/init/__init__.py [hunk 2]
```diff
@@ -80,13 +106,17 @@ def run(self, args: argparse.Namespace):
...
+            self.move_folder()
+            self.update_task_id()
 
-        self.used_tmpdir.cleanup()
+            self.used_tmpdir.cleanup()
 
-        print(util.info(f'Successfully created task "{self.task_id}"'))
+            print(util.info(f'Successfully created task "{self.task_id}"'))
+        except:
+            shutil.rmtree(destination)
```
Comment (MasloMaslane) [inline src/sinol_make/commands/init/__init__.py:121]: Shouldn’t `self.used_tmpdir` also be cleaned up?

**Review by MasloMaslane — CHANGES_REQUESTED** [review]

File: src/sinol_make/commands/init/__init__.py (hunk 1 above)
Comment (j4b6ski) [reply src/sinol_make/commands/init/__init__.py]: I didn't realize other schemes existed 🤔 
Added some more to the list

File: src/sinol_make/commands/init/__init__.py (hunk 2 above)
Comment (j4b6ski) [reply src/sinol_make/commands/init/__init__.py:119]: It should be, thanks
Moved that cleanup into finally block

File: src/sinol_make/commands/init/__init__.py [hunk 3]
```diff
@@ -23,21 +27,32 @@ def configure_subparser(self, subparser: argparse.ArgumentParser):
...
+        print(('Cloning' if is_url else 'Copying') + ' template ' +
+            (f'{subdir} from {template}' if subdir else f'{template}'))
+        if is_url:
+            ret = subprocess.run(['git', 'clone', '-v' if verbose else '-q', '--depth', '1', template, tmpdir])
+            if ret.returncode != 0:
+                util.exit_with_error("Could not access repository. Please try again.")
+            path = os.path.join(tmpdir, subdir)
+        else:
+            path = os.path.join(tmpdir, 'template')
+            shutil.copytree(os.path.join(os.getcwd(), template, subdir), path)
```
Comment (MasloMaslane) [inline src/sinol_make/commands/init/__init__.py:54]: why is there  `os.getcwd()`? Can't `template` be an absolute path?

Comment (MasloMaslane) [inline src/sinol_make/commands/init/__init__.py:54]: why is there `os.getcwd()`? Can't `template` be an absolute path?

**Review by MasloMaslane — CHANGES_REQUESTED** [review]

File: src/sinol_make/commands/init/__init__.py (hunk 3 above)
Comment (j4b6ski) [reply src/sinol_make/commands/init/__init__.py]: seems like os.path.join is surprisingly clever and if template here is an absolute path then os.getcwd() is ignored entirely
image
but os.getcwd() was not required here, so I removed it and added some tests to make sure both absolute and relative paths work

**Review by MasloMaslane — APPROVED** [review]

**Pull request merged**